### PR TITLE
feat: add configuration for enabling RBAC

### PR DESCRIPTION
**Pull request by AlekSi**
Refs #3421.

```yaml
machine:
    # Features describe individual Talos features that can be switched on or off.
    features:
        rbac: true # Enable role-based access control (RBAC).
```

Feature gates are normally on/off toggles, but we want to be able to add additional settings in the future, so I called those "features". But maybe that shouldn't be its own key at all? Just use `machine.rbac.enabled`, `machine.rbac.foo`, etc?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,6 @@ _out
 init.yaml
 controlplane.yaml
 join.yaml
-docgen
```
Comment (AlekSi) [inline .gitignore:10]: Not sure why that was there

File: Dockerfile [hunk 2]
```diff
@@ -102,8 +102,8 @@ ARG STRINGER_VERSION
 RUN go install golang.org/x/tools/cmd/stringer@${STRINGER_VERSION} \
     && mv /go/bin/stringer /toolchain/go/bin/stringer
 RUN curl -sfL https://github.com/uber/prototool/releases/download/v1.10.0/prototool-Linux-x86_64.tar.gz | tar -xz --strip-components=2 -C /toolchain/bin prototool/bin/prototool
-COPY ./hack/docgen /go/src/github.com/talos-systems/docgen
-RUN cd /go/src/github.com/talos-systems/docgen \
+COPY ./hack/docgen /go/src/github.com/talos-systems/talos-hack-docgen
+RUN cd /go/src/github.com/talos-systems/talos-hack-docgen \
```
Comment (AlekSi) [inline Dockerfile:106]: I spent too much time looking for https://github.com/talos-systems/docgen …

File: hack/docgen/main.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ var tpl = `// This Source Code Form is subject to the terms of the Mozilla Publi
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-// DO NOT EDIT: this file is automatically generated by docgen
+// Code generated by hack/docgen tool. DO NOT EDIT.
```
Comment (AlekSi) [inline hack/docgen/main.go:28]: Changed according to https://golang.org/s/generatedcode

File: hack/docgen/main.go [hunk 4]
```diff
@@ -332,10 +334,6 @@ func collectFields(s *structType) (fields []*Field) {
 		yamlTag := tag.Get("yaml")
 		yamlTag = strings.Split(yamlTag, ",")[0]
 
-		if yamlTag == "" {
-			yamlTag = strings.ToLower(yamlTag)
```
Comment (AlekSi) [inline hack/docgen/main.go:336]: Not sure what was the point :)

File: pkg/grpc/middleware/authz/authorizer.go [hunk 5]
```diff
@@ -81,7 +81,7 @@ func (a *Authorizer) authorize(ctx context.Context, method string) (context.Cont
 		return ctx, nil
 	}
 
-	if a.DontEnforce {
+	if a.Enforce {
```
Comment (smira) [inline pkg/grpc/middleware/authz/authorizer.go:84]:
```suggestion
	if !a.Enforce {
```

that seems crucial?

File: pkg/machinery/config/types/v1alpha1/v1alpha1_features.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1
+
+// RBACEnabled implements config.Features interface.
+func (f *FeaturesConfig) RBACEnabled() bool {
+	if f.RBAC == nil {
```
Comment (smira) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_features.go:9]: we should also generate new config for Talos v0.11 with RBAC enabled by default (that's our goal anyways)

File: hack/docgen/main.go (hunk 4 above)
Comment (Unix4ever) [reply hack/docgen/main.go:336]: probably the intention was to do `strings.ToLower(name)`

File: pkg/grpc/middleware/authz/authorizer.go (hunk 5 above)
Comment (AlekSi) [reply pkg/grpc/middleware/authz/authorizer.go]: 🤦 Was debugging unrelated issue and changed that for a quick test…

File: pkg/machinery/config/types/v1alpha1/v1alpha1_features.go (hunk 6 above)
Comment (AlekSi) [reply pkg/machinery/config/types/v1alpha1/v1alpha1_features.go:9]: Done in that PR

File: hack/docgen/main.go (hunk 4 above)
Comment (AlekSi) [reply hack/docgen/main.go:336]: Added a check to require an explicit name

**Comment by AlekSi** [comment]
The proposed config (in the first message) looks good?

**Review by Unix4ever — DISMISSED** [review]

**Review by smira — DISMISSED** [review]
:+1:

**Pull request merged**